api/src/main/java/org/apache/iceberg/catalog/Catalog.java

```diff
@@ -388,6 +392,59 @@ default TableBuilder buildTable(TableIdentifier identifier, Schema schema) {
   default void initialize(String name, Map<String, String> properties) {
   }
 
+  /**
+   * Get catalog properties.
+   *
+   * @return catalog properties
+   */
+  Map<String, String> properties();
```

Comment:
These methods don't need to change `Catalog`. This is something a catalog implementation can define.